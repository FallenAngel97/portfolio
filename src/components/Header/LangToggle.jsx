import React from "react";
import "./LangToggle.scss";
import translate_pic from './translate.svg';

export default class LangToggle extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            langDropDownOpened: false
        }
        this.langClicked = this.langClicked.bind(this);
    }
    langClicked(lang) {
        this.setState({langDropDownOpened: !this.state.langDropDownOpened});
        this.props.langChanged(lang);
    }
    render() {
        return(
            <div onClick={()=>this.setState({langDropDownOpened: !this.state.langDropDownOpened})} id='langToggle'>
                <img src={translate_pic} />
                <span>{this.props.lang}</span>
                <div hidden={!this.state.langDropDownOpened} id='langDropDown'>
                    <span onClick={() => this.langClicked('en') }>EN</span>
                    <span onClick={() => this.langClicked('ru') }>RU</span>
                    <span onClick={() => this.langClicked('uk') }>UA</span>
                </div>
            </div>
        );
    }
}