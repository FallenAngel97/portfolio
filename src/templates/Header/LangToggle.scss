@font-face {
    font-family: 'PT Sans Bold';
    src: url("2223.ttf");
}

#langToggle {
    background: rgba(255,255,255,.05);
    padding: 9px 6px;
    margin-right: 41px;
    transition: .2s ease-in background;
    cursor: pointer;
    position: relative;
    &:hover {
        background: rgba(255,255,255,.16);
    }
    img {
        height: 22px;
        vertical-align: middle;
        margin-right:4px;
    }
    span {
        vertical-align: middle;
        font-family: 'Franklin Gothic Demi';
        color: #e9e9e9;
    }
}
#langDropDown {
    position: absolute;
    width: calc(100% + 24px);
    left: -24px;
    bottom: -25px;
    background: #37373799;
    span {
        margin-right: 10px;
    }
}
.separatorMenu {
    height: 20px;
    width:1px;
    background: #e9e9e9;
    display: inline-block;
    margin-left:10px;
    vertical-align: middle;
    margin-right:10px;
}
#mobileLangToggle {
    background: #515151;
    color: #fbfbfb;
    font-family: 'PT Sans Bold';
    span {
        margin-top: 10px;
        display: inline-block;
        padding: 0px 12px;
    }
}