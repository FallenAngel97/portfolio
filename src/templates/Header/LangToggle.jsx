import React, {useState} from "react";
import "./LangToggle.scss";
import translate_pic from './translate.svg';
import { PortfolioContext } from "../app";

const LangToggle = (props) => {
    const [langDropDownOpened, setLangDropDownOpened] = useState(false)

    const langClicked = (lang) => {
        setLangDropDownOpened((opened) => !opened);
        props.langChanged(lang);
    }

    return(
        <PortfolioContext.Consumer>
            {(isMobile) => (
                <>
                    {isMobile === false ? <div onClick={() => setLangDropDownOpened((opened) => !opened)} id="langToggle">
                        <img src={translate_pic} />
                        <span>{props.lang}</span>
                        <div hidden={!langDropDownOpened} id="langDropDown">
                            <span onClick={() => langClicked('en') }>EN</span>
                            <span onClick={() => langClicked('ru') }>RU</span>
                            <span onClick={() => langClicked('uk') }>UA</span>
                        </div>
                    </div>
                    : <div id="mobileLangToggle">
                        <span onClick={() => langClicked('en') }>EN</span>
                        <span onClick={() => langClicked('ru') }>RU</span>
                        <span onClick={() => langClicked('uk') }>UA</span>
                </div>}
            </>
            )}
        </PortfolioContext.Consumer>
    );
}

export default LangToggle;
