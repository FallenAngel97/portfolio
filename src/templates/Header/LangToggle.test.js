import "../../testSetup";
import React from "react";
import renderer from "react-test-renderer";
import LangToggle from "./LangToggle";

describe('LangToggle element', () => {
    test("LangToggle should render properly", () => {
        const langToogle = renderer.create(<LangToggle  />).toJSON();
        expect(langToogle).toMatchSnapshot();
    });
});