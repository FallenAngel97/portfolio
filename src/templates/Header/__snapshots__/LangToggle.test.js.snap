// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LangToggle element LangToggle should render properly 1`] = `
<div
  id="langToggle"
  onClick={[Function]}
>
  <img
    src="test-file-stub"
  />
  <span />
  <div
    hidden={true}
    id="langDropDown"
  >
    <span
      onClick={[Function]}
    >
      EN
    </span>
    <span
      onClick={[Function]}
    >
      RU
    </span>
    <span
      onClick={[Function]}
    >
      UA
    </span>
  </div>
</div>
`;
